Modify Time

John-JonSteyn opened the pull request:
+ Add increase and Decrease buttons.
+ Expand README.
+ Add LICENSE.

John-JonSteyn committed:
Show Time to count down on timer

~Change the time show to 30'0" rather than 0'0".

John-JonSteyn committed:
Change reset time to WorkMin variable

~ Change the time displayed with the reset value to display the WORK_MIN.

John-JonSteyn committed:
Add requirements.txt

+ Add requirements.txt

John-JonSteyn committed:
Create LICENSE

John-JonSteyn committed:
Update README.md

John-JonSteyn committed:
Optimize code and improve readability

~ Refactor the code to remove redundant comments and blank lines.
~ Update super() calls to use the parentheses-less syntax.
~ Simplify the startBreak() method using a dictionary lookup.
+ Utilize divmod() for calculating minutes and seconds in updateTimeDisplay().
~ Improve variable naming and consistency.
~ Change app ID to version 2.

John-JonSteyn committed:
Move constants to own file

+ Create constants.py
+ Add constants to constants.py
- Remove constants from constants.py

John-JonSteyn committed:
Add increase and decrease buttons

+ Add increaseButton.
+ Add decreaseButton.
~ Style.

John-JonSteyn committed:
Implement + button

+ Add functionality for the + button.

John-JonSteyn committed:
Implement - button

+ Add decreaseButton functionality
+ Cap count down time between 1 and 60

John-JonSteyn committed:
Hide + and - buttons

+ Hide decrease button when timer starts.
+ Hide increase button when timer starts.
+ Show decrease button when timer stops.
+ Show increase button when timer stops.
- Remove comments.

John-JonSteyn committed:
Update main.py

Kayzels reviewed (changes requested):
Due to using PyQt, it can't be MIT license. It requires it to be GPL.

Thanks for pointing out I forgot to add a License.

Kayzels commented on main.py:
```diff
@@ -54,14 +52,14 @@ def updateWork(self):
-            self.modeLabel.setStyleSheet("color: #53baff")
+        break_type = self.workSessions % LONG_BREAK_INTERVAL
+        break_info = {
+            0: ("Long Break", LONG_BREAK_MIN, "#80c342"),
```
Not sure about these changes. Not a fan of random tuples.I agree the original code could be neatened, though. A data lass might be useful.

Also dictionaries that operate as a list also bother me.

John-JonSteyn replied:
Not sure how to do a data lass, but I can find out tomorrow.    : 3

John-JonSteyn replied:
Oh. Class. XD!

Kayzels commented on main.py:
```diff
@@ -6,28 +6,26 @@
         self.setupUi(self)
         self.startButton.clicked.connect(self.startTimer)
         self.resetButton.clicked.connect(self.resetTimer)
-
```
The blank lines here are not redundant. They're there deliberately to break the code in the init into it's logical portions. It makes it easier to follow when skimming.

Call super
Connect button signals to slots
Set timer settings
Set defaults

Kayzels commented on main.py:
```diff
@@ -6,28 +6,26 @@
 class MainWindow(QMainWindow, Ui_MainWindow):
     def __init__(self, *args, **kwargs):
-        super(MainWindow, self).__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
```
This change works, happy here

Kayzels commented on main.py:
```diff
@@ -74,7 +72,7 @@ def updateBreak(self):
     def updateTimeDisplay(self):
         self.timeRemainingSec -= 1
-        mins, secs = secToMinSec(self.timeRemainingSec)
+        mins, secs = divmod(self.timeRemainingSec, 60)
```
Did not know about this function, thank you. Agree with this change.

Kayzels commented on main.py:
```diff
@@ -107,7 +100,7 @@ def run():
     window = MainWindow()
     window.show()
-    app.exec()
+    sys.exit(app.exec())
```
Happy with this change as well

Kayzels commented on main.py:
```diff
@@ -21,10 +14,10 @@ def __init__(self, *args, **kwargs):
         self.resetButton.clicked.connect(self.resetTimer)
         self.workTimer = QTimer()
-        self.workTimer.setInterval(1000) # 1 second
+        self.workTimer.setInterval(1000)
```
The comment here is useful, because it is not immediately clear that the function works in milliseconds.

Kayzels commented:
I'll need to go through more carefully. Some comments that you deleted seemed useful, and should remain. It makes it easier to see the reasoning.

Thank you for your pull request.

John-JonSteyn committed:
Add comments

John-JonSteyn committed:
+ Add Comments

John-JonSteyn committed:
+ Add spaces and comments

John-JonSteyn committed:
Fix Random Tuples

- Remove tuples
+ Add explicit if-else statements to determine break type in startBreak method.
- Remove unnecessary code for setting stylesheet.

John-JonSteyn committed:
Implement continuous increase / decrease functionality

+ Add methods startIncrease and startDecrease to handle continuous increase and decrease functionalities when the corresponding buttons are held down.
+ Start timers (increaseTimer and decreaseTimer) to call the respective continuousIncrease and continuousDecrease methods at regular intervals while the buttons are held down.
+ Stop the timers when the buttons are released.

John-JonSteyn committed:
Optimise

~ Reorganized imports for better readability
+ Implement separate button press/release signals for increase and decrease buttons
+ Add methods to handle continuous increase and decrease while buttons are held down
~ Optimise the updateTimeDisplay() method for better performance

John-JonSteyn committed:
Set Max Min to 90

~ Change max minutes to 90.

Kayzels commented:
Sorry for closing this, and not doing more with it. And for taking so long to get back to you.

Unfortunately, this request makes too many changes, so I would need to cherry pick and go through to select the ones I want. This seems to be more effort than it's worth, for a project that I haven't used in years. It also appears that you have a main branch in your fork with an updated version that does make changes such as correcting the license, but there wasn't a pull request from that branch.
